### PR TITLE
Revert #13975

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Reverts #13975 after concerns raised in https://github.com/conda/conda/issues/12558#issuecomment-2245517955.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: news/14065-revert-13975 [hunk 1]
```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* <news item>
+
+### Bug fixes
+
+* Revert potential regression introduced in #13975. The previously forbidden characters are now (`^`, `%`, `!`, `=`, `(`, `)`, `\`) allowed again til the impact is assessed. (#14065)
```
Comment (kenodegard) [inline news/14065-revert-13975:7]:
```suggestion
* Revert potential regression introduced in #13975. The previously forbidden characters (`^`, `%`, `!`, `=`, `(`, `)`, `\`) are allowed again until the impact is assessed. (#14065)
```

**Review by kenodegard — DISMISSED** [review]

**Comment by jaimergp** [comment]
Those benchmarks 🚀  hahah

**Review by beeankha — APPROVED** [review]

**Comment by mbargull** [comment]
@jezdez pointed out in https://github.com/conda/conda/issues/12558#issuecomment-2245951439 that my feedback is too late in the release process (with which I agree).
I don't know the formalities of the process and as such can't gauge what's more adequate between blocking an in-process release vs putting out a hot fix release after (neither is ideal, of course).
I agree with this feedback:
> I’d suggest to work on an improved patch that only presents users with a note that future versions of conda will change env name handling.

**Comment by mbargull** [comment]
@jezdez suggested to put out a warning that we'll reiterate on this in the future.
I've pushed https://github.com/conda/conda/pull/14073 as a suggestion to give a rough idea on how we could do that.

**Review by kenodegard — APPROVED** [review]

**Pull request merged**